lib/node/message.ml

```diff
@@ -5,15 +5,27 @@ type category =
   | Uncategorized
   | Request
   | Response
+  | Post
   | Failure_detection
   | Custom            of string
-[@@deriving bin_io]
+[@@deriving bin_io, show]
 
 type t = {
   category : category;
   id : int;
+  timestamp : float;
   sender : Address.t;
-  recipient : Address.t;
+  recipients : Address.t list;
   payload : bytes;
 }
 [@@deriving bin_io]
+
+let hash_of m =
+  [
+    m.sender.address;
+    string_of_int m.sender.port;
+    string_of_float m.timestamp;
+    Bytes.to_string m.payload;
```

Comment:
Should we add the `id`?
Same question about recipients, but I have a strong feeling that it is not useful.